core/src/main/java/com/segment/analytics/kotlin/core/Analytics.kt

```diff
@@ -99,27 +102,31 @@ open class Analytics protected constructor(
     // This function provides a default state to the store & attaches the storage and store instances
     // Initiates the initial call to settings and adds default system plugins
     internal fun build() {
-        // because startup queue doesn't depend on a state, we can add it first
-        add(StartupQueue())
-        add(ContextPlugin())
-        add(UserInfoPlugin())
-
-        // Setup store
-        analyticsScope.launch(analyticsDispatcher) {
-            store.also {
-                // load memory with initial value
-                it.provide(userInfo)
-                it.provide(System.defaultState(configuration, storage))
+        try {
+            // because startup queue doesn't depend on a state, we can add it first
+            add(StartupQueue())
+            add(ContextPlugin())
+            add(UserInfoPlugin())
+
+            // Setup store
+            analyticsScope.launch(analyticsDispatcher) {
+                store.also {
+                    // load memory with initial value
+                    it.provide(userInfo)
+                    it.provide(System.defaultState(configuration, storage))
+
+                    // subscribe to store after state is provided
+                    storage.subscribeToStore()
+                }
 
-                // subscribe to store after state is provided
-                storage.subscribeToStore()
-            }
+                if (configuration.autoAddSegmentDestination) {
+                    add(SegmentDestination())
+                }
 
-            if (configuration.autoAddSegmentDestination) {
-                add(SegmentDestination())
+                checkSettings()
             }
-
-            checkSettings()
+        } catch (t: Throwable) {
```

Comment:
maybe we don't have to include analyticsScope.launch in the try block, since the scope has its own handler?

Reply:
Good callout. Moved this into the timeline.add() function which is what ends up calling the plugin.setup().